packages/common/src/util/base64.js

```diff
@@ -1,26 +1,63 @@
+import _ from 'lodash';
+
 /**
- * Encodes a given string into Base64 format.
+ * Encodes a given string or Javascript object into Base64 format.
  * @function
  * @public
  * @namespace util
- * @param {string} data - The string to be encoded.
+ * @param {string | object} data - The string or object to be encoded.
+ * @param {object} options - Options.
  * @returns {string} - The Base64 encoded string.
  * @example <caption>Encode a string</caption>
- * const encoded = encode('Hello World');
+ * const encodedString = encode('Hello World');
  * console.log(encoded); // Output: SGVsbG8gV29ybGQ=
+ * @example <caption>Encode an object</caption>
+ * const encodedObject = encode({name: 'Jane Doe'})
+ * console.log(encodedObject); //output eyJuYW1lIjoiSmFuZSBEb2UifQ==
+ * @example <caption>To skip the JSON stringification step</caption>
+ * const encodedObject = encode('Hello World', {parseJson: false})
  */
-export const encode = data => Buffer.from(data, 'utf-8').toString('base64');
+export const encode = (data, options = {parseJson: true}) => {
+  let str = data;
 
-/**
+  if(typeof data !== "string" && options.parseJson){
+    try {
+      str = JSON.stringify(str);
+    } catch (e){
+      console.log(e.message);
+    }
+  }
+
+  return Buffer.from(str, 'utf-8').toString('base64');
+}
+
+/**4
  * Decodes a Base64 encoded string back to its original format.
  * @function
  * @public
  * @namespace util
  * @param {string} base64Data - The Base64 encoded string.
- * @returns {string} - The decoded string.
+ * @param {object} options - Options.
+ * @returns {string | object} - The decoded string or JavaScript Object.
  * @example <caption>Decode a Base64 string</caption>
  * const decoded = decode('SGVsbG8gV29ybGQ=');
- * console.log(decoded); // Output: Hello World
+ * @example <caption>Decode a Base64 JSON object to a standard JavaScript object</caption>
+ * const decoded = decode('eyJuYW1lIjoiSmFuZSBEb2UifQ==');
+ * console.log(decoded); // Output: {name: 'Jane Doe'}
+ * @example <caption>To skip the JSON stringification step</caption>
+ * const decodedString = decode('Hello World', {parseJson: false})
  */
-export const decode = base64Data =>
-  Buffer.from(base64Data, 'base64').toString('utf-8');
+export const decode = (base64Data, options = {parseJson: true}) =>{
+  let decodedObject = Buffer.from(base64Data, 'base64').toString('utf-8');
+
+  if((_.startsWith(decodedObject, '[') || _.startsWith(decodedObject, '{')) && options.parseJson) {
```

Comment:
Oh, just for the record String has a `startsWith` function - so you can just do `decodedObject.startsWith(['])`

For readability I would probably put `options.parseJson` first. It doesn't _really_ matter but there's no point parsing the string contents if we're not going to use the result anyway :)